src/pudl/extract/eia_aeo.py

```diff
@@ -0,0 +1,225 @@
+                f"table_{table_number}_"
+            )
+        }
+        matching_series = set(
+            itertools.chain.from_iterable(
+                self.graph.nodes[n_id].get("childseries", [])
+                for n_id in matching_category_ids
+            )
+        )
+
+        series_records = itertools.chain.from_iterable(
+            self.__series_to_records(series_id, potential_parents=matching_category_ids)
+            for series_id in matching_series
+        )
+        return pd.DataFrame.from_records(series_records)
+
+
+@multi_asset(
+    outs={
+        "raw_eia_aeo__electric_power_projections_regional": AssetOut(is_required=False),
+        "raw_eia_aeo__renewable_energy_regional": AssetOut(is_required=False),
+    },
+    can_subset=True,
+    required_resource_keys={"datastore"},
+)
+def raw_eia_aeo(context: AssetExecutionContext):
+    """Extract tables from EIA's Annual Energy Outlook.
+
+    We first extract a taxonomy from the AEO JSON blob, which connects
+    individual data series to "categories". Some categories are associated with
+    a specific table; others are associated with an AEO case or subject.
+
+    The AEO cases are different scenarios such as "High Economic Growth" or
+    "High Oil Price." They include "Reference" and "2022 AEO reference case" as
+    well.
+
+    The AEO subjects are only used for filtering which tables are relevant to
+    which subjects, e.g. "Table 54 is relevant to Energy Prices." So we ignore
+    those right now.
+
+    The series each have their own timeseries data, as well as some metadata
+    such as a series name and units. Many different dimensions can be inferred
+    from the series names, but the data is somewhat heterogeneous so we do not
+    try to infer those here and leave that to the transformation step.
+    """
+    name_to_number = {
+        "raw_eia_aeo__electric_power_projections_regional": 54,
+        "raw_eia_aeo__renewable_energy_regional": 56,
+    }
+    ds = context.resources.datastore
+    year = 2023
+    filename = f"AEO{year}.txt"
+
+    with ds.get_zipfile_resource("eia_aeo", year=year).open(
+        filename, mode="r"
+    ) as aeo_raw:
+        taxonomy = AEOTaxonomy(io.TextIOWrapper(aeo_raw))
+
+    selected = context.op_execution_context.selected_output_names
+    for asset_name in selected:
+        yield Output(
+            value=taxonomy.get_table(name_to_number[asset_name]), output_name=asset_name
+        )
+
+
+@asset_check(asset="raw_eia_aeo__electric_power_projections_regional")
+def raw_table_54_invariants(df: pd.DataFrame) -> AssetCheckResult:
+    """Check that the AEO Table 54 raw data conforms to *some* assumptions.
+
+    * all values are non-null - i.e. every fact has date, fact name, category
+    name, case, and unit
+    * we have values from every electricity market module region
+    * covers 20 cases and 26 electricity market module regions (25 regions + 1 national)
```

Comment:
I think the lack of a blank line after this bulleted list is what's breaking the docs.